### PR TITLE
Ridley Combat Logic Overhaul

### PR DESCRIPTION
**Pull request by Samuel6710**

### CODE REVIEW

**Comment by JeffGainsNGames** [comment]
+120k diff

**Comment by Samuel6710** [comment]
May have added py-randomprime to the folder

File: randovania/games/prime1/logic_database/Tallon Overworld.txt [hunk 1]
```diff
@@ -616,13 +616,28 @@ Extra - aabb: [-576.05975, -221.71631, -188.35965, -170.50539, 225.37242, 45.510
       Trivial
   > Event - Meta Ridley
       All of the following:
-          Artifact of Chozo and Artifact of Elder and Artifact of Lifegiver and Artifact of Nature and Artifact of Newborn and Artifact of Spirit and Artifact of Strength and Artifact of Sun and Artifact of Truth and Artifact of Warrior and Artifact of Wild and Artifact of World and Shoot Any Beam
-          Charge Beam or Combat (Intermediate)
+          Have all Artifacts
```
Comment (JeffGainsNGames) [inline randovania/games/prime1/logic_database/Tallon Overworld.txt:619]: You are missing a change in the header file to define this new template. I imagine you lost the change somewhere when recreating the branch.

**Review by JeffGainsNGames — CHANGES_REQUESTED** [review]

**Review by Miepee — CHANGES_REQUESTED** [review]
The health logic for ridley seems totally scuffed, would even say that its worse. It basically causes "you have charge beam, now go fight ridley hitless, gl" after combat beginner.

Try seperating this into:
```
ALL
 # Phase 1 Item requiements
 # Phase 2 Item requirements
 # Phase 3 Item requirements
 # Total health requirements
 ```

**Comment by Samuel6710** [comment]
I'm going to redo these changes from the ground up since Miepee makes some good points about why it's flawed and it could cause a lot of problems.

**Pull request closed without merging**